src/spectplan.jl

```diff
@@ -1,91 +1,86 @@
-- `pad_rot{padu_rot, padd_rot, padl_rot, padr_rot}` padded pixels for {left,right,up,down} direction for image rotation
-- `ncore` number of CPU cores used to process data, must be integer
+- `nthread` number of CPU threads used to process data, must be integer
+- `planrot` Vector of struct `PlanRotate`
+- `planpsf` Vector of struct `PlanPSF`
 Currently code assumes the following:
 * each of the `nview` projection views is `[nx,nz]`
 * `nx = ny`
 * uniform angular sampling
 * `psf` is symmetric
-* multiprocessing using # of cores specified by `Threads.nthreads()`
+* multiprocessing using # of threads specified by `Threads.nthreads()`
 """
-struct SPECTplan
+
+struct SPECTplan{T}
     T::DataType # default type for work arrays etc.
-    imgr::AbstractArray{<:RealU, 3} # 3D rotated image, (nx, ny, nz)
-    add_img::AbstractArray{<:RealU, 3}
-    mumap::AbstractArray{<:RealU, 3} # [nx,ny,nz] attenuation map, must be 3D, possibly zeros()
-    mumapr::AbstractArray{<:RealU, 3} # 3D rotated mumap, (nx, ny, nz)
-    psfs::AbstractArray{<:RealU, 4} # PSFs must be 4D, [nx_psf, nz_psf, ny, nview], finally be centered psf
-    nview::Int # number of views
-    viewangle::AbstractVector{<:RealU}
-    interpidx::Int
-    dy::RealU
     imgsize::NTuple{3, Int}
-    psfsize::NTuple{2, Int}
-    pad_fft::NTuple{4, Int}
-    pad_rot::NTuple{4, Int}
-    ncore::Int # number of cores
+    nx_psf::Int
+    imgr::Array{T, 3} # 3D rotated image, (nx, ny, nz)
+    add_img::Array{T, 3}
+    mumap::Array{T, 3} # [nx,ny,nz] attenuation map, must be 3D, possibly zeros()
+    mumapr::Array{T, 3} # 3D rotated mumap, (nx, ny, nz)
+    exp_mumapr::Vector{Array{T, 2}} # 2D exponential rotated mumap, (nx, ny)
+    psfs::Array{T, 4} # PSFs must be 4D, [nx_psf, nz_psf, ny, nview], finally be centered psf
+    nview::Int # number of views
+    viewangle::StepRangeLen{T}
+    interpmeth::Symbol
+    dy::T
+    nthread::Int # number of threads
+    planrot::Vector{PlanRotate}
+    planpsf::Vector{PlanPSF}
     # other options for how to do the projection?
-    function SPECTplan(
-        mumap::AbstractArray{<:RealU, 3},
-        psfs::AbstractArray{<:RealU, 4},
-        dy::RealU;
-        viewangle::AbstractVector{<:RealU} = (0:size(psfs, 4) - 1) / size(psfs, 4) * (2π), # set of view angles
-        interpidx::Int = 2, # 1 is for 1d interpolation, 2 is for 2d interpolation
-        T::DataType = promote_type(eltype(mumap), Float32),
-    )
+
+    function SPECTplan(mumap::Array{<:RealU, 3},
+                       psfs::Array{<:RealU, 4},
+                       dy::RealU;
+                       T::DataType = promote_type(eltype(mumap), Float32),
+                       viewangle::StepRangeLen{<:RealU} = (0:size(psfs, 4) - 1) / size(psfs, 4) * T(2π), # set of view angles
+                       interpmeth::Symbol = :two, # :one is for 1d interpolation, :two is for 2d interpolation
+                       nthread::Int = Threads.nthreads(),
+                       )
```

Comment:
```suggestion
    function SPECTplan(
        mumap::Array{<:RealU, 3},
        psfs::Array{<:RealU, 4},
        dy::RealU;
        T::DataType = promote_type(eltype(mumap), Float32),
        viewangle::StepRangeLen{<:RealU} = (0:size(psfs, 4) - 1) / size(psfs, 4) * T(2π), # set of view angles
        interpmeth::Symbol = :two, # :one is for 1d interpolation, :two is for 2d interpolation
        nthread::Int = Threads.nthreads(),
    )
```

hopefully i did this with 4 space indent - if not please fix.  hard to tell on github!